src/FunctionApps/PITest_FunctionApp/tests/test_decrypt_function.py

```diff
@@ -77,3 +78,74 @@ def test_decrypt_message_failure_wrong_receiver(local_settings):
         )
 
     assert "Cannot decrypt the provided message with this key" in str(exc_info.value)
+
+
+def test_trigger_success(local_settings):
+    """Test function trigger conditions
+
+    Args:
+        local_settings ([type]): passed automatically via above fixture
+    """
+    test_file_path = (
+        files("DecryptFunction").parent / "tests" / "assets" / "encrypted.txt"
+    )
+    blob_data = test_file_path.read_bytes()
+    input_stream = func.blob.InputStream(data=blob_data, name="input test")
+
+    req_success = func.HttpRequest(
+        method="POST",
+        body=input_stream.read(),
+        headers={"Content-Type": "application/octet-stream"},
+        url="/",
+    )
+    resp = dcf.main(req_success, local_settings)
+    assert resp.status_code == 200
+    assert resp.get_body() == b"TESTING EICR ENCRYPTION"
+
+
+def test_trigger_missing_body(local_settings):
+    """Test trigger with missing body
+
+    Args:
+        local_settings ([type]): passed automatically via above fixture
+    """
+    req_success = func.HttpRequest(
+        method="POST",
+        body="",
+        headers={"Content-Type": "application/octet-stream"},
+        url="/",
+    )
+    resp = dcf.main(req_success, local_settings)
+    assert resp.status_code == 400
+
+
+def test_trigger_malformed(local_settings):
+    """Test trigger with malformed bytes
+
+    Args:
+        local_settings ([type]): passed automatically via above fixture
+    """
+    req_success = func.HttpRequest(
+        method="POST",
+        body=b"bad data",
+        headers={"Content-Type": "application/octet-stream"},
+        url="/",
+    )
+    resp = dcf.main(req_success, local_settings)
+    assert resp.status_code == 500
```

Comment:
Can we test that it's failing for the right reason? There are a couple ways a status code of 500 is returned, so I think it would be better to test that the status code is 500 **and** the response is "decryption failed".